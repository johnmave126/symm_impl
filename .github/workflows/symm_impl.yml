name: Test

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v1

      - name: Install rust toolchain
        uses: actions-rs/toolchain@v1
        with:
          toolchain: stable
          override: true
          default: true

      - name: Restore Cache
        uses: actions/cache@v2
        with:
          path: |
            ~/.cargo/registry
            ~/.cargo/git
            target
          key: test-cargo-${{ hashFiles('**/Cargo.toml') }}

      - name: Quick check
        run: cargo check

      - name: Test
        run: cargo test