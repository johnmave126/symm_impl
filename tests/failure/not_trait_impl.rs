use symm_impl::symmetric;

struct A {
    a: i32,
}

#[symmetric]
impl A {
    fn new(a: i32) -> Self {
        Self { a }
    }
}

fn main() {}
