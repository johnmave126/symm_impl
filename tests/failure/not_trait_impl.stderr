error: #[symmetric] can only be used on trait implementation
 --> $DIR/not_trait_impl.rs:7:1
  |
7 | #[symmetric]
  | ^^^^^^^^^^^^
  |
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
